react/TextLink/TextLink.demo.js

```diff
@@ -1,27 +1,19 @@
-import React from 'react';
-import PropTypes from 'prop-types';
 import TextLink from './TextLink';
 import Text from '../Text/Text';
 
-const TextLinkContainer = ({ component: DemoComponent, componentProps }) => (
-  <Text>
-    <DemoComponent {...componentProps} />
-  </Text>
-);
-TextLinkContainer.propTypes = {
-  component: PropTypes.any,
-  componentProps: PropTypes.object.isRequired
-};
+import { sizes } from '../private/withTextProps';
+import capitalize from 'lodash/capitalize';
 
 export default {
   route: '/text-link',
   title: 'Text Link',
   category: 'Typography',
   component: TextLink,
-  container: TextLinkContainer,
   initialProps: {
     href: '#',
-    children: 'Text link'
+    children: 'Text link',
+    component: Text,
```

Comment:
Not sure we should be showing this as a standard usage pattern. It seems like `TextLink` is designed to be nested within a `Text` component, so maybe we make the demo show this `Text > TextLink` nesting instead? 🤔

Reply:
If we want to do this, we'd probably need to tweak the way demos work, since this works quite differently to existing demos.

Reply:
That feels weird that it's meant to be nested in Text, whilst you could supply Text as a component as well, thus ending up with two nested Text components...plus in case of just wrapping you can't pass the props of the wrapper down 🤔  Obviously if something like anchor tag is wrapped that's okay. If it's meant to be wrapped in Text why don't we just include this Text as an essential part of this component 🤔  hm...

Reply:
Might need to chat about this to figure out how to make the API handle this cleanly, it's a bit confusing.